hikari/events/poll_events.py

```diff
@@ -0,0 +1,104 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Events related to polls."""
+
+from __future__ import annotations
+
+__all__: typing.Sequence[str] = ("PollVoteAdd", "PollVoteRemove")
+
+import typing
+
+import attrs
+
+from hikari.events import shard_events
+from hikari.internal import attrs_extensions
+
+if typing.TYPE_CHECKING:
+    from hikari import snowflakes
+    from hikari import traits
+    from hikari.api import shard as gateway_shard
+
+
+@attrs_extensions.with_copy
+@attrs.define(kw_only=True, weakref_slot=False)
+class PollVoteAdd(shard_events.ShardEvent):
+    """Event that is fired when a user add their vote to a poll.
+
+    If the poll allows multiple selection, one event will be fired for each vote.
+    """
+
+    app: traits.RESTAware = attrs.field(metadata={attrs_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from Event>>.
+
+    shard: gateway_shard.GatewayShard = attrs.field(metadata={attrs_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    user_id: snowflakes.Snowflake = attrs.field()
+    """ID of the user that added their vote to the poll."""
+
+    channel_id: snowflakes.Snowflake = attrs.field()
+    """ID of the channel that the poll is in."""
+
+    message_id: snowflakes.Snowflake = attrs.field()
+    """ID of the message that the poll is in."""
+
+    guild_id: typing.Optional[snowflakes.Snowflake] = attrs.field(default=None)
+    """ID of the guild that the poll is in.
+
+    This will be [None][] if the poll is in a DM channel.
+    """
+
+    answer_id: int = attrs.field()
+    """ID of the answer that the user voted for."""
+
+
+@attrs_extensions.with_copy
+@attrs.define(kw_only=True, weakref_slot=False)
+class PollVoteRemove(shard_events.ShardEvent):
+    """Event that is fired when a user remove their vote to a poll.
+
+    If the poll allows multiple selection, one event will be fired for each vote.
+    """
+
+    app: traits.RESTAware = attrs.field(metadata={attrs_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from Event>>.
+
+    shard: gateway_shard.GatewayShard = attrs.field(metadata={attrs_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    user_id: snowflakes.Snowflake = attrs.field()
+    """ID of the user that removed their vote from the poll."""
+
+    channel_id: snowflakes.Snowflake = attrs.field()
+    """ID of the channel that the poll is in."""
+
+    message_id: snowflakes.Snowflake = attrs.field()
+    """ID of the message that the poll is in."""
+
+    guild_id: typing.Optional[snowflakes.Snowflake] = attrs.field(default=None)
+    """ID of the guild that the poll is in.
+
+    This will be [None][] if the poll is in a DM channel.
+    """
+
+    answer_id: int = attrs.field()
+    """ID of the answer that the user remove their vote from."""
```

Comment:
I have a feeling `answer_index` is the better naming for this

Reply:
Fair enough, just thought it might be nicer when it comes to actually knowing what `answer_id` refers to